diff --git a/packages/contracts-gen/src/DsourceEmpty.client.ts b/packages/contracts-gen/src/DsourceEmpty.client.ts
index bbbd9d5..abbf9df 100644
--- a/packages/contracts-gen/src/DsourceEmpty.client.ts
+++ b/packages/contracts-gen/src/DsourceEmpty.client.ts
@@ -65,6 +65,22 @@ export interface DsourceEmptyInterface extends DsourceEmptyReadOnlyInterface {
     owner: string;
   }, $fee?: number | StdFee | "auto", $memo?: string, $funds?: Coin[]) => Promise<ExecuteResult>;
 }
+/**
+ * ```ts
+ * const { contractAddress } = await client.deploy(
+   admin,
+   path.join(testDataDir, 'dsource_empty.wasm'),
+   {
+     language: 'node',
+     script_url: 'https://gist.githubusercontent.com/tubackkhoa/4ab5353a5b44118ccd697f14df65733f/raw/4a27d2ac4255d23463286898b161eda87d1b95bb/datasource_coingecko.js',
+     parameters: ['ethereum']
+   } as DsourceEmptyTypes.InstantiateMsg,
+   'dsource',
+   'auto'
+ );
+ const dsourceContract = new DsourceEmptyClient(client, admin, contractAddress);
+ * ```
+ */
 export class DsourceEmptyClient extends DsourceEmptyQueryClient implements DsourceEmptyInterface {
   client: SigningCosmWasmClient;
   sender: string;
